mypy_django_plugin/main.py

```diff
@@ -313,7 +325,10 @@ def get_dynamic_class_hook(self, fullname: str) -> Optional[Callable[[DynamicCla
 
     def report_config_data(self, ctx: ReportConfigContext) -> Dict[str, Any]:
         # Cache would be cleared if any settings do change.
-        return self.plugin_config.to_json()
+        extra_data = {}
+        if ctx.id == "django.contrib.auth":
+            extra_data["AUTH_USER_MODEL"] = self.django_context.settings.AUTH_USER_MODEL
+        return self.plugin_config.to_json(extra_data)
```

Comment:
Now we get "implicit" cache tainting on `django.contrib.auth`, and thus `authenticate` and friends if `AUTH_USER_MODEL` changes.

There might be other modules we want to add some extra config data to. Not sure though